library/core/src/ffi/c_str.rs

```diff
@@ -207,6 +212,22 @@ impl fmt::Display for FromBytesWithNulError {
     }
 }
 
+#[unstable(feature = "ub_checks", issue = "none")]
+impl Invariant for &CStr {
+    /**
+     * Safety invariant of a valid CStr:
+     * 1. An empty CStr should have a null byte.
+     * 2. A valid CStr should end with a null-terminator and contains
+     *    no intermediate null bytes.
+     */
+    fn is_safe(&self) -> bool {
+        let bytes: &[c_char] = &self.inner;
+        let len = bytes.len();
+
+        return !bytes.is_empty() && bytes[len - 1] == 0 && !bytes[..len-1].contains(&0);
```

Comment:
One more nit :)
```suggestion
        !bytes.is_empty() && bytes[len - 1] == 0 && !bytes[..len-1].contains(&0)
```

Reply:
Thank you. I still haven't gotten use to this Rust magic

Reply:
Fixed